akka-projection-jdbc/src/main/scala/akka/projection/jdbc/javadsl/JdbcHandler.scala

```diff
@@ -40,7 +40,7 @@ object JdbcHandler {
  * implementation.
  */
 @ApiMayChange
-abstract class JdbcHandler[Envelope, S <: JdbcSession] extends JdbcHandlerLifecycle {
+trait JdbcHandler[Envelope, S <: JdbcSession] extends JdbcHandlerLifecycle {
```

Comment:
I think the reason for it being an abstract class might be that the `akka.projection.javadsl.Handler` is an abstract class. For Handler there is a good reason since it has default implementation of some methods.

Yes, we don't have to align the JdbcHandler, but I'm not sure it's worth the change? It can often be solved by delegation/composition instead.

Reply:
> Yes, we don't have to align the JdbcHandler, but I'm not sure it's worth the change? 

I tried to use the handler as the lambda and delegate the processing of envelopes to another class. It looks not bad, thank you, Patrik. 👍  As Russian says "The morning is wiser than the evening" 😄 

But I got a problem with signature a `JdbcHandler.fromFunction` function and I think that need to change it to more useful in Java code. I changed PR for that, WDYT?